### DISCUSSION TITLE
Migrate to 4.5.2 failed

### DISCUSSION THREAD

**Discussion by JoeFox82** [nextgenhealthcare/connect#6338] [Q&A]
Hi,

I don't know if it is my fault or it is a bug. I migrate from Version 4.5.1 to 4.5.2 and the server doesn't start. I took a look in the logs and saw that there is an error in line 40 of class com.mirth.connect.server.migration.Migrate4_5_2. I checked the code and it tells me:

`builder = PropertiesConfigurationUtil.createBuilder(new File(ClassPathResource.getResourceURI("log4j2-cli.properties")));`

The problem is that I do not have a file _log4j2-cli.properties_ in my conf folder. So I copied the log4j2.properties and saved them as log4j2-cli.properties. After restarting Mirth server it works now.

Is that a bug? Should there be a check whether the file exists or not?

Please note. I had the same problem when updating from 4.4.2 to 4.5.2 on another instance!

Kind regards

**Comment by pacmano1** [comment]
~~I suspect that is local to your install.  Perhaps a permission issue.~~

See further down this thread.  

Also, you post here often enough that you need to include what operating system and all that jazz you're on.

**Reply by JoeFox82** [reply]
I'm sorry for that. I did the migration from 4.5.1 to 4.5.2 on an Windows 10 pro system and I'm a local admin. The migration from 4.4.2 to 4.5.2 was done on an Windows Server 2019 Standard with a local admin. For both systems, mirth connect is running as a service under the local system account.

This morning I tried it on a local windows 11 pro system from 4.4.2 to 4.5.2 as a local admin, service is running as local system user. Same error:
 
> ERROR 2024-10-01 10:16:15.335 [Main Server Thread] Server: Exception in thread "Main Server Thread" java.lang.NullPointerException: Cannot invoke "java.net.URI.isAbsolute()" because "uri" is null
> ERROR 2024-10-01 10:16:15.343 [Main Server Thread] Server: 	at java.base/java.io.File.<init>(File.java:417)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.migration.Migrate4_5_2.migrateLog4jProperties(Migrate4_5_2.java:40)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.migration.Migrate4_5_2.migrate(Migrate4_5_2.java:23)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.migration.ServerMigrator.migrate(ServerMigrator.java:78)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.controllers.DefaultMigrationController.migrate(DefaultMigrationController.java:96)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.Mirth.startup(Mirth.java:319)
> ERROR 2024-10-01 10:16:15.344 [Main Server Thread] Server: 	at com.mirth.connect.server.Mirth.run(Mirth.java:168)
 
Edit: All installations are stored at the default directory C:\Program files\Mirth Connect

**Reply by ab-mg-23** [reply]
I can't repro on a test system. My configuration:

Windows Server 2K22
Amazon Corretto 11.0.21
Clean install of Mirth 4.4.2 w/x64 installer, verified log4j2-cli.properties existed in conf folder.
Upgrade install to Mirth 4.5.2 w/x64 installer, verified log4j2-cli.properties still existed in conf folder.

**Reply by JoeFox82** [reply]
I do not have a log4j2-cli.properties in the conf folder after the initial installation of Mirth 4.4.2! I think that is the main problem.

**Reply by ab-mg-23** [reply]
I can reproduce if I deselect the Mirth Connect CLI component during the 4.4.2 install. As is expected, the installer doesn't install the log4j2-cli.properties file.

**Reply by JoeFox82** [reply]
You are right, I did not install the CLI component

**Reply by pacmano1** [reply]
Interesting - so if you don't install the cli the upgrade process does not work.  Seems like a bug to report in the Issues section.   It's a good catch.

**Reply by JoeFox82** [reply]
I've created an issue https://github.com/nextgenhealthcare/connect/issues/6341
